.github/workflows/ft_tests.yml

```diff
@@ -0,0 +1,39 @@
+name: Free Threading Tests
+on: [push, pull_request]
+jobs:
+  tests:
+    runs-on: ubuntu-latest
+    container:
+        image: ghcr.io/nascheme/numpy-tsan:3.13t
+        options: --shm-size=2g # increase memory for large matrix ops
+    steps:
+      - uses: actions/checkout@v4
+      - name: Install Dependencies
+        run: |
+          set -x
+          set -e
+          python -m pip install packaging pyflakes pytest-cov pytest-flakes
+          # Test dep, consider skipping
+          python -m pip install sympy hypothesis
+          # NumPy and Cython are already at the right versions for this image
+          # python -m pip install --pre --upgrade -i https://pypi.anaconda.org/scientific-python-nightly-wheels/simple cython
+          # python -m pip install --pre --upgrade -i https://pypi.anaconda.org/scientific-python-nightly-wheels/simple numpy
+          python -mpip install pytest-run-parallel
+      - name: Test Installation
+        run: |
+          python -m pip install . --no-build-isolation -Csetup-args=-Db_sanitize=thread -Csetup-args=-Dbuildtype=debugoptimized -v
```

Comment:
It doesn't look like ndindex uses meson, so this won't build the ndindex extension using TSAN either.